web/index_handler.go

```diff
@@ -61,28 +63,55 @@ func (ac *assetCache) asset(path string) (string, error) {
 func IndexHandler(logger lager.Logger, publicFS fs.FS, live bool) http.Handler {
 	assets := newAssetCache(publicFS, live)
 
+	funcs := template.FuncMap{
+		"asset": assets.asset,
+		"jsonMarshal": func(v any) (template.JS, error) {
+			payload, err := json.Marshal(v)
+			if err != nil {
+				return "", err
+			}
+			return template.JS(payload), nil
+		},
+	}
+
+	// Cache template in production
+	var cachedTemplate *template.Template
+	if !live {
+		var err error
+		cachedTemplate, err = template.New("web").Funcs(funcs).ParseFS(publicFS, "index.html")
+		if err != nil {
+			panic(fmt.Sprintf("failed to parse index.html: %v", err))
+		}
+	}
+
+	bufPool := sync.Pool{
+		New: func() interface{} {
+			return new(bytes.Buffer)
+		},
+	}
+
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		log := logger.Session("index")
 
-		funcs := template.FuncMap{
-			"asset": assets.asset,
-			"jsonMarshal": func(v any) (template.JS, error) {
-				payload, err := json.Marshal(v)
-				if err != nil {
-					return "", err
-				}
-				return template.JS(payload), nil
-			},
-		}
+		var t *template.Template
+		var err error
 
-		t, err := template.New("web").Funcs(funcs).ParseFS(publicFS, "index.html")
-		if err != nil {
-			log.Error("failed-to-parse-templates", err)
-			w.WriteHeader(http.StatusInternalServerError)
-			return
```

Comment:
I'm trying to understand what was wrong with the previous code and how you would get 200 OK on template errors. From what I can see here it appears we return HTTP 500 if the template errors.

Do you have a reproducible or any logs from this log line `failed-to-parse-templates`?